core/dbt/adapters/base/impl.py

```diff
@@ -1125,6 +1125,37 @@ def get_rows_different_sql(
 
         return sql
 
+    def validate_incremental_strategy(self):
+        return ["append", "delete+insert"]
```

Comment:
Lots of thoughts here — thanks for pushing forward the conversation!

- I think this should be `append` by default—the truly common denominator—but then Postgres (`plugins/postgres/dbt/adapters/posrtgres/impl.py`) should reimplement as both `append` + `delete+insert`
- Where does this validation get called? Within `get_incremental_strategy_macro`?
- This works great for adapter maintainers to override. As an average user, though, how would I add my new strategy `silly`? Should our presumption be that, if I've defined a macro named `get_incremental_silly_sql` (i.e. in a non-internal package), I want to use it? That would work, but feels like tricky implicit behavior

Reply:
If it only validates the "builtin"s, and trusts the user for anything else, I'm cool with that! As long as there's a clear error message if they've specified `incremental_strategy: 'silly'`, and we can't find a macro named `get_incremental_silly_sql`